### PR TITLE
tests: update spread tests to account for content snaps

### PR DESCRIPTION
**Pull request by cjp256**
tests: update gnome-3-28 extension spread test to use gtk

tests: update gnome extension spread task to account for content snaps

tests: update kde extension spread task to account for content snaps

### CODE REVIEW

**Comment by cjp256** [comment]
For future use, I PR'd a NOMATCH helper for spread to simplify those checks.

File: tests/spread/extensions/gnome-3-28/task.yaml [hunk 1]
```diff
@@ -28,18 +32,32 @@ restore: |
   . "$TOOLS_DIR/snapcraft-yaml.sh"
   restore_yaml "snap/snapcraft.yaml"
 
+  # Remove content snaps installed by test.
```
Comment (sergiusens) [inline tests/spread/extensions/gnome-3-28/task.yaml]: This is not needed, it is done in the general restore (`restore-each` in `spread.yaml`)

File: tests/spread/extensions/gnome-3-28/task.yaml [hunk 2]
```diff
@@ -12,13 +12,17 @@ systems:
...
 environment:
-  SNAP_DIR: ../snaps/glib-hello
+  SNAP_DIR: ../snaps/gtk3-hello
 
 prepare: |
   #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
   . "$TOOLS_DIR/snapcraft-yaml.sh"
   set_base "$SNAP_DIR/snap/snapcraft.yaml"
 
+  # Remove content snaps to be installed by test.
```
Comment (sergiusens) [inline tests/spread/extensions/gnome-3-28/task.yaml]: This really shouldn't be needed, I suspect this returning 0 may even be a bug and will hit us in the future.

File: tests/spread/extensions/kde-neon/task.yaml [hunk 3]
```diff
@@ -16,6 +16,9 @@ prepare: |
   . "$TOOLS_DIR/snapcraft-yaml.sh"
   set_base "$SNAP_DIR/snap/snapcraft.yaml"
 
+  # Remove content snaps to be installed by test.
```
Comment (sergiusens) [inline tests/spread/extensions/kde-neon/task.yaml]: ditto

File: tests/spread/extensions/kde-neon/task.yaml [hunk 4]
```diff
@@ -16,6 +16,9 @@ prepare: |
...
+  sudo snap remove kde-frameworks-5-core18
```
Comment (sergiusens) [inline tests/spread/extensions/kde-neon/task.yaml]: ditto

File: tests/spread/extensions/kde-neon/task.yaml [hunk 5]
```diff
@@ -25,9 +28,12 @@ restore: |
   . "$TOOLS_DIR/snapcraft-yaml.sh"
   restore_yaml "snap/snapcraft.yaml"
 
+  # Remove content snaps installed by test.
```
Comment (sergiusens) [inline tests/spread/extensions/kde-neon/task.yaml]: ditto

File: tests/spread/extensions/gnome-3-28/task.yaml (hunk 1 above)
Comment (cjp256) [reply tests/spread/extensions/gnome-3-28/task.yaml]: Thanks!

**Comment by cjp256** [comment]
> Just remove the logic that removes, we should expect a clean environment already.
> 
> If you want to be sure, grep $output and verify the snaps were installed or just check if they are installed in `prepare`, we will have a bigger problem if the snap was installed prior

Ah, I did realize `restore-each` purged all the snaps. Fixed up unnecessary removes! Thanks!

**Pull request merged**